File: src/my_project/easy_problems/from201to250/consecutive_strings.py
class Solution:
    def maxPower(self, s: str) -> int:
        
        # initiolize variables
        len_s: int = len(s)
        max: int  = 0
        count: int = 0
        
        # compute the maximum length of the substring with only one character
        for i in range(len_s):
            count = 1
            for j in range(i+1, len_s):
                if s[i] == s[j]:
                    count += 1
                else: 
                    break
            
            
            if count > max: 
                max = count
                            
        return max
    
    
solution= Solution()

print(solution.maxPower(s = "leetcode"))

print(solution.maxPower(s = "abbcccddddeeeeedcba"))

print(solution.maxPower(s = "triplepillooooow"))

print(solution.maxPower(s = "hooraaaaaaaaaaay"))

print(solution.maxPower(s = "tourist"))

File: my_project_typescript/easy_problems_typescript/from101to150/consecutiveStrings.ts
function maxPower(s: string): number {

    // initialize variables
    let lenS: number = s.length
    let maxNumber: number = 0
    let count: number = 0

    // compute the maximum length of the substring with only one character
    for (let i = 0; i < lenS; i++){
        count = 1
        for (let j = i+1; j < lenS; j++){
            if (s[i] === s[j]){
                count += 1
            }
            else {
                break
            }
        }
        if (count > maxNumber){
            maxNumber = count 
        }
    }

    return maxNumber 

};

console.log(maxPower("leetcode"))

console.log(maxPower("abbcccddddeeeeedcba"))

console.log(maxPower("triplepillooooow"))

console.log(maxPower("hooraaaaaaaaaaay"))

console.log(maxPower("tourist"))
